cyclops-ui/src/components/pages/ModuleDetails.tsx

```diff
@@ -144,6 +145,32 @@ const ModuleDetails = () => {
 
   let { moduleName } = useParams();
 
+  const [kind, setKind] = useState<string>("");
+  const [showManagedFields, setShowManagedFields] = useState(false);
+
+
+  const handleCheckboxChange = (e: CheckboxChangeEvent) => {
+    setShowManagedFields(e.target.checked);
+  };
+
+
+  useEffect(() => {
+    resources.forEach((resource: any, index) => {
+      if(resource.kind === kind){
+        fetchManifest(
+          resource.group,
+          resource.version,
+          resource.kind,
+          resource.namespace,
+          resource.name,
+        );
+      }
+    })
+  },[showManagedFields]);
```

Comment:
I wouldn't go over all resources we have for a module until I find the one we need.

You can add `group`, `version`, `kind`, `name` and `namespace` of the resource to the `manifestModal` state when the Manifest button is clicked. This way you don't have to go over all the resources and just get it from the state.

So basically what you did with the `kind` state, but you can pack it into the same state as `manifestModal` and add all of the fields so you don't have to iterate over resources